web/src/beta/lib/reearth-ui/components/NumberInput/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useCallback, useEffect, FC } from "react";
+
+import { fonts, styled } from "@reearth/services/theme";
+
+export type NumberInputProps = {
+  value?: number;
+  size?: "normal" | "small";
+  disabled?: boolean;
+  placeholder?: string;
+  appearance?: "readonly";
+  unit?: string;
+  onChange?: (value?: number | string) => void;
+  onBlur?: (value?: number | string) => void;
+};
+
+export const NumberInput: FC<NumberInputProps> = ({
+  value,
+  size = "normal",
+  disabled,
+  placeholder,
+  appearance,
+  unit,
+  onChange,
+  onBlur,
+}) => {
+  const [currentValue, setCurrentValue] = useState<number | string | undefined>(value);
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    setCurrentValue(value);
+  }, [value]);
+
+  const handleChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const currentValue = e.currentTarget.value;
+      if (/^-?\d*\.?\d*$/.test(currentValue)) {
+        setCurrentValue(currentValue);
+        onChange?.(parseFloat(currentValue));
+      }
+    },
+    [onChange],
+  );
+
+  const handleBlur = useCallback(() => {
+    setIsFocused(false);
+    onBlur?.(currentValue);
+  }, [onBlur, currentValue]);
+
+  const handleFocus = useCallback(() => {
+    setIsFocused(true);
+  }, []);
+
+  return (
+    <Wrapper size={size} status={isFocused ? "active" : "default"}>
+      <StyledInput
+        type="text"
+        inputMode="numeric"
+        pattern="[0-9]*"
```

Comment:
Do we still need these since we already handle the validate with regexp?

Reply:
@airslice removed it